### DISCUSSION TITLE
Python 3.10 MacOS  numpy dependent wheels

### DISCUSSION THREAD

**Discussion by Czaki** [pypa/cibuildwheel#847] [Q&A]
I try to build wheels for package which uses Numpy. For windows and Linux it works perfect. On MacOS it ends with `RankWarning: Polyfit may be poorly conditioned`. As  read It is caused by lack of proper BLAS library. But using `brew installl openblas` does not solve this. Did you have any hint how to deal with it?

@henryiii  ?

**Comment by henryiii** [comment]
I believe there's a config setting for the build that has to be set. See https://github.com/MacPython/numpy-wheels to see how they do it. env_vars.sh has it, I think.

**Reply by Czaki** [reply]
I do not find in this repo way to build. I will wait on NumPy wheels. U found that also matplotlib people do not found https://github.com/matplotlib/matplotlib/pull/20819 a way to solve it.

**Reply by henryiii** [reply]
I've just left testing off for macOS & windows 3.10, personally, since it will be much slower too.

**Reply by Czaki** [reply]
Windows 3.10 works. On macOS there is also a problem with pp3.7

**Comment by cmccully** [comment]
Likely related, this is specifically for pp3.7. It appears that numpy as of 1.19 and higher requires std=c11 instead of c99. I have confirmed that for a local install of pp3.7 numpy 1.18 compiles but 1.19 and higher does not. 

This is the important part of the crash log:

```
/tmp/pypy3.7-v7.3.5-osx64/include/pyport.h:68:20: error: redefinition of typedef 'Py_hash_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
typedef Py_ssize_t Py_hash_t;
                   ^
/tmp/pypy3.7-v7.3.5-osx64/include/cpyext_object.h:9:14: note: previous definition is here
typedef long Py_hash_t;
             ^
1 error generated.
error: Command "gcc -pthread -arch x86_64 -DNDEBUG -O2 -std=c11 -fPIC -DNPY_INTERNAL_BUILD=1 -DHAVE_NPY_CONFIG_H=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -Inumpy/core/include -Ibuild/src.macosx-10.7-x86_64-3.7/numpy/core/include/numpy -Ibuild/src.macosx-10.7-x86_64-3.7/numpy/distutils/include -Ibuild/src.macosx-10.7-x86_64-3.7/numpy/core/src/umath -Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/tmp/pypy3.7-v7.3.5-osx64/include -Ibuild/src.macosx-10.7-x86_64-3.7/numpy/core/src/common -Ibuild/src.macosx-10.7-x86_64-3.7/numpy/core/src/npymath -c build/src.macosx-10.7-x86_64-3.7/numpy/core/src/umath/_umath_tests.dispatch.avx2.c -o build/temp.macosx-10.7-x86_64-3.7/build/src.macosx-10.7-x86_64-3.7/numpy/core/src/umath/_umath_tests.dispatch.avx2.o -MMD -MF build/temp.macosx-10.7-x86_64-3.7/build/src.macosx-10.7-x86_64-3.7/numpy/core/src/umath/_umath_tests.dispatch.avx2.o.d -std=c99 -Werror -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mavx2" failed with exit status 1
```
I've replaced `-std=c99` with `-std=c11` in this gcc command by hand and then things compile (at least) successfully. 

For reference, this is the repo I am trying to build wheels with: 
https://github.com/astropy/astroscrappy/runs/4027187559?check_suite_focus=true

Any ideas how to get around this @henryiii ? I may have to force pp3.7 to simply use an older version of numpy.
